### PR TITLE
[FLINK-28151]Allow to cancel the Operation for the HiveServer2 Endpoint.

### PR DESCRIPTION
**Pull request by chaozwn**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 1]
```diff
@@ -123,6 +133,56 @@ public void testGetException() throws Exception {
...
+        assertThat(tCancelOperationResp.getStatus().getStatusCode())
+                .isEqualTo(TStatusCode.SUCCESS_STATUS);
+    }
+
+    @Test
+    public void testCloseOperation() throws Exception {
+        TCLIService.Client client = createClient();
+        TOpenSessionReq openSessionReq = new TOpenSessionReq();
+        TOpenSessionResp openSessionResp = client.OpenSession(openSessionReq);
+        SessionHandle sessionHandle =
```
Comment (fsk119) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:166]: Actually we can use SqlGatewayService#openSession.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 2]
```diff
@@ -123,6 +133,56 @@ public void testGetException() throws Exception {
...
+                                "select 1",
+                                0,
+                                Configuration.fromMap(Collections.emptyMap()));
+        TOperationHandle tOperationHandle =
+                toTOperationHandle(sessionHandle, operationHandle, OperationType.EXECUTE_STATEMENT);
+        TCancelOperationReq tCancelOperationReq = new TCancelOperationReq(tOperationHandle);
+        TCancelOperationResp tCancelOperationResp =
+                ENDPOINT_EXTENSION.getEndpoint().CancelOperation(tCancelOperationReq);
+        assertThat(tCancelOperationResp.getStatus().getStatusCode())
+                .isEqualTo(TStatusCode.SUCCESS_STATUS);
```
Comment (fsk119) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java:158]: We also need to verify the operation is canceled.

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I left some comments.

**Comment by fsk119** [comment]
I add a commit to fix the mentioned problem. Could you take a look when CI test passes, @chaozwn ?

**Comment by chaozwn** [comment]
> I add a commit to fix the mentioned problem. Could you take a look when CI test passes, @chaozwn ?

@fsk119 yes,i do.

**Comment by chaozwn** [comment]
> I add a commit to fix the mentioned problem. Could you take a look when CI test passes, @chaozwn ?

LGTM

**Review by fsk119 — APPROVED** [review]
Thanks for the response. LGTM.

**Pull request closed without merging**